### PR TITLE
use isinstance() to determine type rather than []

### PR DESCRIPTION
**Pull request by xuelanghanbao**
#1376 
Use `isinstance()` to determine `tup[0]` type rather than `tup[0]==[]`. When `tup` is passed in from `self.w.append(array_ops_compat.hstack(wi))` in `deepxde\data\fpde.py, line 474`, `tup[0]==[]` can cause the following errors: `ValueError: operands could not be broadcast together with shapes (172,) (0,)`

### CODE REVIEW

**Comment by lululxvi** [comment]
Good point. But here we want to check if it is a empty list, not only checking if it is a list.

**Comment by xuelanghanbao** [comment]
Thank you for your explanation. I have a better understanding of the purpose of this code. So changing `not is_tensor(tup[0])` to `isinstance(tup[0],list)` would be better, right?

**Comment by lululxvi** [comment]
> changing `not is_tensor(tup[0])` to `isinstance(tup[0],list)` would be better, right?

Seems OK, but I am not 100% sure. It would be safer to use both.

File: deepxde/utils/array_ops_compat.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def convert_to_array(value):
 
 
 def hstack(tup):
-    if not is_tensor(tup[0]) and tup[0] == []:
+    if not is_tensor(tup[0]) and isinstance(tup[0],list) and tup[0] == []:
```
Comment (lululxvi) [inline deepxde/utils/array_ops_compat.py:25]: Format via black https://pypi.org/project/black/

**Pull request merged**